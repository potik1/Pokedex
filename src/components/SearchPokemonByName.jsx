import React, {Component} from 'react';
import {connect} from 'react-redux';
import {searchPokemon } from '../actions/index';

class SearchPokemonByName extends Component {

  constructor(props) {
    super(props);
    this.state = {term: ''};
  }

  onInputChange(event) {
    this.setState({term: event.target.value});
  }

  onFormSubmit(event) {
    event.preventDefault();
    this.props.searchPokemon(this.state.term);
    this.setState({term:''});
  }

  render() {
    return (
      <form
        onSubmit={(event)=> this.onFormSubmit(event)}
        className="form-inline my-2 my-lg-0 ">
        <input
          placeholder="Enter pokemon name..."
          className="form-control"
          value={this.state.term}
          onChange={(event) => this.onInputChange(event)}
        />
        <span className="input-group-btn ">
               <button type="submit"
                       className="btn btn-secondary">Submit</button>
          </span>
      </form>
    );
  }
}

export default connect(null, {searchPokemon})(SearchPokemonByName);
