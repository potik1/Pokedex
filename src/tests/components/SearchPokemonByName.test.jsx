import React from 'react';
import { shallow } from 'enzyme';
import SearchPokemonByName from '../../components/SearchPokemonByName';

describe('Search Pokemon', () => {
  const searchPokemon = shallow(<SearchPokemonByName />);

  test('renders correctly', () => {
    expect(searchPokemon).toMatchSnapshot();
  });
});
