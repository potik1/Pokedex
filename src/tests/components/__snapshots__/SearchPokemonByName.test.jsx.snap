// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Pokemon renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPokemonByName
    searchPokemon={[Function]}
    term=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter pokemon name..."
          value=""
/>,
        <span
          className="input-group-btn "
>
          <button
                    className="btn btn-light"
                    type="submit"
          >
                    Search
          </button>
</span>,
      ],
      "className": "form-inline my-2 my-lg-0 ",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "onChange": [Function],
          "placeholder": "Enter pokemon name...",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-light"
            type="submit"
>
            Search
</button>,
          "className": "input-group-btn ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "btn btn-light",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
        "type": "span",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Enter pokemon name..."
            value=""
/>,
          <span
            className="input-group-btn "
>
            <button
                        className="btn btn-light"
                        type="submit"
            >
                        Search
            </button>
</span>,
        ],
        "className": "form-inline my-2 my-lg-0 ",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "onChange": [Function],
            "placeholder": "Enter pokemon name...",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-light"
              type="submit"
>
              Search
</button>,
            "className": "input-group-btn ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "btn btn-light",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
          "type": "span",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
